src/dask_awkward/lib/core.py

```diff
@@ -711,6 +742,7 @@ def _partitions(self, index: Any) -> Array:
 
         # if a single partition was requested we trivially know the new divisions.
         if len(raw) == 1 and isinstance(raw[0], int) and self.known_divisions:
+            # TODO: don't we always know the divisions?
```

Comment:
I know uproot sometimes doesn't report the sizes of the partitions (I think it can make the `uproot.dask` call a bit faster to ignore determining the sizes of the partitions), and if we have something read from JSON we don't know the divisions (only npartitions)